pypesto/hierarchical/solver.py

```diff
@@ -298,3 +296,74 @@ def fun(x):
             x_opt = scale_value_dict(x_opt, problem)
 
         return x_opt
+
+    def sample_startpoints(
+        self, problem: InnerProblem, pars: List[InnerParameter]
+    ) -> np.ndarray:
+        """Sample startpoints for the numerical optimization.
+
+        Samples the startpoints for the numerical optimization from a
+        log-uniform distribution using the symmetric logarithmic scale.
+
+        Parameters
+        ----------
+        problem:
+            The inner problem to solve.
+        pars:
+            The inner parameters to sample startpoints for.
+
+        Returns
+        -------
+        The sampled startpoints appended to the cached startpoints.
+        """
+        if self.minimize_kwargs['n_starts'] == 1 and self.x_guesses is None:
+            return np.array(
+                [list(problem.get_dummy_values(scaled=False).values())]
+            )
+        elif self.x_guesses is not None:
+            n_samples = self.minimize_kwargs['n_starts'] - len(self.x_guesses)
+        else:
+            n_samples = self.minimize_kwargs['n_starts'] - 1
+
+        if n_samples <= 0:
+            return self.x_guesses
+
+        lb = np.array(
+            [x.lb if x.lb != -np.inf else self.dummy_lb for x in pars]
+        )
+        ub = np.array(
+            [x.ub if x.ub != np.inf else self.dummy_ub for x in pars]
+        )
+
+        def symlog(x):
+            return np.sign(x) * np.log10(np.abs(x) + 1)
+
+        def inverse_symlog(x):
+            return np.sign(x) * (np.power(10, np.abs(x)) - 1)
+
+        # Sample startpoints from a log-uniform distribution
+        startpoints = np.random.uniform(
+            low=symlog(lb),
+            high=symlog(ub),
+            size=(n_samples, len(pars)),
+        )
+        startpoints = inverse_symlog(startpoints)
+
+        # Stack the sampled startpoints with the cached startpoints
+        if self.x_guesses is not None:
+            startpoints = np.vstack(
+                (
+                    self.x_guesses,
+                    startpoints,
+                )
+            )
+        else:
+            startpoints = np.vstack(
+                (
+                    np.array(
+                        [list(problem.get_dummy_values(scaled=False).values())]
+                    ),
```

Comment:
Why are the dummy values only used in the case where the user doesn't supply `x_guesses`? I think it makes sense to use.

If so, then the highlighted code is currently used at a two points in the code, so it could be computed in advance (doesn't reduce computation, just lines of code).